homeassistant/components/switchbot/sensor.py

```diff
@@ -0,0 +1,79 @@
+"""Support for SwitchBot sensors."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DATA_COORDINATOR, DOMAIN, MANUFACTURER, SensorType
+from .coordinator import SwitchbotDataUpdateCoordinator
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbot sensor based on a config entry."""
+    coordinator: SwitchbotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    sensors = []
+
+    if coordinator.data[entry.unique_id].get("data"):
+        for sensor in coordinator.data[entry.unique_id].get("data"):
+            if sensor in SensorType.__members__:
+                sensor_type = getattr(SensorType, sensor).value
+                sensors.append(
+                    SwitchBotSensor(
+                        coordinator,
+                        entry.unique_id,
+                        sensor,
+                        sensor_type,
+                        entry.data[CONF_MAC],
+                        entry.data[CONF_NAME],
+                    )
+                )
+
+    async_add_entities(sensors)
+
+
+class SwitchBotSensor(CoordinatorEntity, Entity):
+    """Representation of a Switchbot sensor."""
+
+    coordinator: SwitchbotDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: SwitchbotDataUpdateCoordinator,
+        idx: str | None,
+        sensor: str,
+        sensor_type: str,
+        mac: str,
+        switchbot_name: str,
+    ) -> None:
+        """Initialize the Switchbot sensor."""
+        super().__init__(coordinator)
+        self._idx = idx
+        self._sensor = sensor
+        self._mac = mac
+        self._attr_unique_id = f"{self._mac.replace(':', '')}-{sensor}"
```

Comment:
Since this happens quite a few places in the integration, it would be good to make a helper to format it instead of repeating the `.replace` everywhere

Reply:
IDX is equal to self._mac.replace(':', '')

I just replaced this with IDX